### DISCUSSION TITLE
File Folder Permission Proposal

### DISCUSSION THREAD

**Discussion by goxiaoy** [directus/directus#5248] [Feature Requests]
The current file management and permission come complicate when handles some use cases like

- Some files like user's picture avatar, images uploaded in WYSIWYG or Markdown are public,  no access token needed, and cache friendly for some application (keep the same url when user's access_token changes)
- Some other files like user's uploaded would fiter by role or user id

Maybe it's better to have ability to manage file permission by folder

1. Organize file into folders like 
    * 'user/avatar'
    * 'post/picture'
    
    I have seen https://github.com/directus/directus/pull/3209 which provide such ability

2. Manage permission by folder (powered by filter)
    * 'user/avatar': public read, user can only create any, unable to delete/update
    * 'post/picture': public read, subset of roles can write
    
    filter by such syntax
```
{
...
  "folder":{
    "path":{
      "_in":["user/avatar","'post/picture"]
    }
  }
...
}
```

**Comment by rijkvanzanten** [comment]
> Maybe it's better to have ability to manage file permission by folder

You can already do this, by using the IDs of the folders:

```json
{
  "folder": {
    "_in": [
      "fac21847-d5ce-4e4b-a288-9abafbdfbc87",
      "a5bdb793-dd85-4ac9-882a-b42862092983"
    ]
  }
}
```

**Reply by justrealmilk** [reply]
Like this?

![image](https://user-images.githubusercontent.com/156681/128564964-0c93087f-a041-41ec-81a2-487de606f6df.png)

**Reply by rijkvanzanten** [reply]
@justrealmilk Yeah, this should give the public permission read access to files where the folder field equals one of those two IDs 👍🏻

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._v
